src/components/Editor/EditorFileModalInterface.tsx

```diff
@@ -45,11 +45,17 @@ const indexName = `${
 
 const FileSearch = ({
   onSelect,
+  openAddFile,
 }: {
   onSelect: (file: AlgoliaEditorFile) => void;
+  openAddFile: () => void;
 }) => {
   const { query, refine: setQuery } = useSearchBox();
   const { hits } = useHits() as { hits: AlgoliaEditorFileHit[] };
+  console.log(hits.map(hit => hit.__position));
+  if (!hits.find(hit => hit.id === 'none')) {
+    hits.push({ id: 'none' } as AlgoliaEditorFileHit); // blank hit
+  }
```

Comment:
Instead of doing this, maybe it's neater to just add another Button after `hits.map` for "Add New Problem".

Reply:
That actually might be slightly more annoying, since styling will have to unnecessarily copied (if someone in the future wants to modify it, they'll have two things to modify...); gonna keep it like this for now!

Reply:
Hmm the styling being unnecessarily copied is a good point, but I think in this case it's still better to have the duplication:

- The duplication is fairly short; it's not a large amount of duplication
- The duplication is right next to each other and it's obvious that it's duplicated, so making changes to both at the same time (ex. if we wanted to change some styles) is pretty fast (just ctrl+d to make multiple cursors on VSCode). Tailwind CSS wrote a blog post about this at some point (duplicated HTML classes was fine because of multiple cursors or something)
- Without the duplication, there's extra logic that's not immediately obvious. In particular, the onClick() handler now needs a separate branch for `if (hit.id === 'none')`, and it's not immediately obvious why that's there and what it's doing. With the duplication, the logic can be different and it's very clear why (one button is handling a normal hit and the other button is adding an option to create a new solution). We don't need any special checks.
- It's also strange that hits[] now has an extra item at the end of it. Imagine one day we wanted to make a "hits" counter in the popup -- we would have to remember to subtract 1 from the length of hits[].
- creating a new solution does not always have to have the same styling as the rest of the hits. For example, imagine we wanted to add a "plus" icon next to create new solution -- with the current approach, we would have to add another `hits.id === 'none'` check to conditionally render the icon. Or, we might want to make the new solution button have static position in the modal.

If you really didn't want to duplicate the button though, maybe you could change `hits.id` from `none` to `CREATE_NEW_SOLUTION` or something to make it super obvious what that's for? and / or change `// blank hit` to a comment explaining that this is a hit placeholder used to show a create new solution button. Or, you could refactor the button into a new component, but I think this is probably overkill.

(If you strongly want to keep it as is though, that's fine, I would just be curious as to why -- I haven't thought too much about this so maybe there's something I missed)

Reply:
icic, thanks for the detailed explanation; will be changing this!